File: src/Entity/Sponsor.php
<?php

namespace App\Entity;

use App\Repository\SponsorRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: SponsorRepository::class)]
class Sponsor extends User
{

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: 'Company name is required.')]
    #[Assert\Length(max: 255, maxMessage: 'Company name cannot be longer than {{ limit }} characters.')]
    private ?string $companyName = null;

    #[ORM\Column(length: 255, nullable: true)]
    #[Assert\Length(max: 255, maxMessage: 'Website URL cannot be longer than {{ limit }} characters.')]
    #[Assert\Url(message: 'Please enter a valid website URL.')]
    private ?string $website = null;

    #[ORM\Column(length: 255, nullable: true)]
    #[Assert\Length(max: 255, maxMessage: 'Address cannot be longer than {{ limit }} characters.')]
    private ?string $address = null;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $description = null;

    #[ORM\Column(options: ['default' => false])]
    private ?bool $verified = false;

    public function getCompanyName(): ?string
    {
        return $this->companyName;
    }

    public function setCompanyName(string $companyName): static
    {
        $this->companyName = $companyName;

        return $this;
    }

    public function getWebsite(): ?string
    {
        return $this->website;
    }

    public function setWebsite(?string $website): static
    {
        $this->website = $website;

        return $this;
    }

    public function getAddress(): ?string
    {
        return $this->address;
    }

    public function setAddress(?string $address): static
    {
        $this->address = $address;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function isVerified(): ?bool
    {
        return $this->verified;
    }

    public function setVerified(bool $verified): static
    {
        $this->verified = $verified;

        return $this;
    }
}

File: Java_Application/src/main/java/com/pegasus/entities/Sponsor.java
package com.pegasus.entities;

import java.time.LocalDateTime;

public class Sponsor extends User {
    private String companyName;
    private String website;
    private String address;
    private String description;
    private boolean verified;

    public Sponsor() {
    }

    public Sponsor(
            String email,
            String roles,
            String password,
            String username,
            String phone,
            String avatarUrl,
            LocalDateTime createdAt,
            String status,
            String dtype,
            String companyName,
            String website,
            String address,
            String description,
            boolean verified
    ) {
        super(email, roles, password, username, phone, avatarUrl, createdAt, status, dtype);
        this.companyName = companyName;
        this.website = website;
        this.address = address;
        this.description = description;
        this.verified = verified;
    }

    public Sponsor(
            Integer id,
            String email,
            String roles,
            String password,
            String username,
            String phone,
            String avatarUrl,
            LocalDateTime createdAt,
            String status,
            String dtype,
            String companyName,
            String website,
            String address,
            String description,
            boolean verified
    ) {
        this(email, roles, password, username, phone, avatarUrl, createdAt, status, dtype, companyName, website, address, description, verified);
        setId(id);
    }

    public String getCompanyName() {
        return companyName;
    }

    public void setCompanyName(String companyName) {
        this.companyName = companyName;
    }

    public String getWebsite() {
        return website;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public boolean isVerified() {
        return verified;
    }

    public void setVerified(boolean verified) {
        this.verified = verified;
    }
}
